src/shared_utilities/utils.jl

```diff
@@ -666,6 +684,27 @@ function NaNCheckCallback(
     SciMLBase.DiscreteCallback(cond, affect!)
 end
 
+# no date version
```

Comment:
Could you explain a little more here? Why don't we use this method in all cases?